packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -188,8 +188,9 @@ class FlutterDevice {
       }
       printTrace('Successfully connected to service protocol: $observatoryUri');
 
-      vmServices = <VMService>[service];
-      device.getLogReader(app: package).connectedVMServices = vmServices;
+      vmService = service;
+      service.device = device;
```

Comment:
This is where the vmservice learns about the device type.

Reply:
Could you pass the device to `VMService.connect` ?